doc/guides/using-internal-errors.md

```diff
@@ -0,0 +1,112 @@
+# Using the internal/errors.js Module
+
+## What is internal/errors.js
+
+The `require('internal/errors')` module is an internal-only module that can be
+used to produce `Error`, `TypeError` and `RangeError` instances that use a
+static, permanent error code and an optionally parameterized message. The
+intent of the module is to make errors thrown by Node.js core to be more
+consistent and to allow changes to error messages to be made without requiring
+a semver-major label.
+
+Switching an existing error to use the `internal/errors` module must be
+considered a semver-major change. However, once using `internal/errors`,
+changes to `internal/errors` error messages will be handled as semver-minor
+or patch.
+
+## Using internal/errors.js
+
+The `internal/errors` module exposes three custom `Error` classes as a
+`message` function.
+
+### Class: errors.Error(key[, args...])
```

Comment:
Since we now have rest parameters, I’d be in favour of using `...args` in these places if that’s possible?